Tasks/SonarQubePostTest/SummaryReport/QualityGateSection.ps1

```diff
@@ -0,0 +1,222 @@
+
+function GetQualityGateSectionContent
+{
+    Write-Verbose "Formatting the quality gate report section"
+    
+    WaitForAnalysisToFinish
+    
+    $qualityGateResponse = FetchQualityGateDetails
+    $qualityGateStatus = $qualityGateResponse.projectStatus.status
+    $section1 = FormatQualityGateStatusSection $qualityGateStatus
+    
+    $warningsAndErrors = GetQualityGateWarningsAndErrors $qualityGateResponse    
+    $section2 = FormatQualityGateMessagesSection $warningsAndErrors
+    
+    $content = FormatQualityGateSection $section1 $section2
+    return $content
+}
+
+function FormatQualityGateSection
+{
+    param ($section1, $section2)
+    
+    return "$section1 $section2"
+}
+
+function FormatQualityGateMessagesSection
+{
+    param ($messages)
+    
+    if (HasElements $messages)
+    {
+        Write-Host "here"
+        $sb = New-Object "System.Text.StringBuilder"
+        [void]$sb.AppendLine('<table border="0" style="border-top: 1px solid #eee;border-collapse: separate;border-spacing: 0 2px;">')
+        
+        foreach ($message in $messages)
+        {
+            Write-Verbose "Value $($message.actualValue)"
+            
+            [void]$sb.AppendLine("<tr>")
+            [void]$sb.AppendLine("<td><span style=""padding-right:4px;"">$($message.metric_name)</span></td>")
+            [void]$sb.AppendLine("<td style=""text-align: center; background-color:$($message.color); color:#fff;""><span style=""padding:0px 2px"">$($message.actualValue)</span></td>")
+            [void]$sb.AppendLine('<td>&nbsp;' + $message.comparator + ' ' + $message.threshold + '</td>')
+            [void]$sb.AppendLine("</tr>")
+        }
+        
+        [void]$sb.AppendLine("</table>")
+    } 
+    
+    return $sb.ToString()
```

Comment:
It looks like $sb won't be initialised unless there are message (line 33). Won't this call fail in that case?